Add demo pipeline that builds Reader for ASC distribution

mokagio opened the pull request:
Builds on top of https://github.com/wordpress-mobile/WordPress-iOS/pull/24439 because that PR has a prototype solution for decoupling Keystone from the secrets and allowing release builds to pass. Otherwise, the `GenerateCredentials.sh` would fail because when building Keystone as a pre-requisite for Reader because there are no secrets available for that target.

The end goal is to distribute to TestFlight, but this is a good intermediate step we can merge in the meantime.

<img width="926" alt="image" src="https://github.com/user-attachments/assets/dee114ab-3c37-4e41-991f-b097b927eb44" />

mokagio committed:
Inherit from matching common xcconfig in debug and release

Before, they both inherited from alpha.

mokagio committed:
Configure `PROVISIONING_PROFILE_SPECIFIER` for Reader debug and release

Ignoring the alpha (Enterprise distribution) build for the moment.

mokagio committed:
Add lane to build Reader for App Store Connect distribution

mokagio committed:
DRY shared setup to build apps in CI

mokagio committed:
Add CI step to build Reader — Build only for the moment

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Update access control of some methods necessary for Keystone to build

See how
https://buildkite.com/automattic/wordpress-ios/builds/27253/steps?sid=01961f84-fc91-442d-89c1-84f330f7b000
failed.

mokagio commented on WordPress/Classes/Models/Media.swift:
```diff
@@ -9,14 +9,14 @@ extension Media {
     ///
     @objc
-    func incrementAutoUploadFailureCount() {
+    public func incrementAutoUploadFailureCount() {
```
This and the other changes were necessary for Keystone to compile. See how https://buildkite.com/automattic/wordpress-ios/builds/27253/steps?sid=01961f84-fc91-442d-89c1-84f330f7b000 failed.

mokagio commented on config/Reader.debug.xcconfig:
```diff
@@ -1,6 +1,8 @@
-#include "Common.alpha.xcconfig"
+#include "Common.debug.xcconfig"
```
@kean I assumed the debug and release `xcconfig` including the alpha common settings was a copy pasta. But let me know if it was intentional and we'll find a way to adjust. Thanks!

mokagio commented on fastlane/lanes/build.rb:
```diff
@@ -266,6 +266,22 @@
     )
   end
 
+  lane :build_for_app_store_connect_reader do
+    sentry_check_cli_installed
+
+    update_certs_and_profiles_app_store_reader
```
Like the lane to fetch the cert and provisioning profile, this lane has "reader" as the suffix. I'll follow up at some point to adjust the names of all lanes so they are consistent and tab-completion friendly.

With that I mean

- `build_for_...`
- `build_for_firebase_...
- `build_for_app_store_connect_...`
- `build_for_app_store_connect_jetpack`
- `build_for_app_store_connect_wordpress`
- `build_for_app_store_connect_reader`

But I need to take renames slowly because I need to check usages in Rv2.
